### DISCUSSION TITLE
How to disable parameters being prefilled with autocomplete?

### DISCUSSION THREAD

**Discussion by voidtask** [zed-industries/zed#15735] [Feature Requests]
Whenever I use autocomplete in Zed to complete Go functions I end up getting all parameters prefilled and those also for some reason have types specified, how to disable it? I don't want anything inside of the function's call braces to appear at all 

![image](https://github.com/user-attachments/assets/56050706-aac9-4367-90de-88ca21d76041)

**Comment by cv4x** [comment]
I would guess this is the `completeFunctionCalls` setting in gopls: https://github.com/golang/tools/blob/master/gopls/doc/settings.md#completefunctioncalls-bool 

Maybe this can be set from within settings.json under the "lsp" property?

Edit: found the answer here: https://github.com/zed-industries/zed/issues/4660#issuecomment-2058972379

Try this:
```json
    "lsp": {
        "gopls": {
            "initialization_options": {
                "completeFunctionCalls": false
            }
        }
    }
```

**Reply by voidtask** [reply]
Thanks! Although it's not exactly solving the problem, but it certainly is way better than having to deal with all those prefilled parameters! I think that the problem itself could be a bug of zed tho? Not too sure about that honestly, at least I can definitely say that the behavior I'm experiencing is not happening neither in neovim, nor vscode. Anyway, thank you very much for the solution!

**Comment by mandarliev** [comment]
Do you know how to do the same for javascript?

**Comment by grian32** [comment]
Little necroposting, but disabling complete function calls leads to the behaviour where it only autocompletes function names and not parens aswell, you can achieve the usual behaviour from other IDE's where it just autocompletes parens and puts your cursor in them with these settings, the setting that actually adds in those function signatures is `usePlaceholders`, which should be false by default.
```json
  "lsp": {
    "gopls": {
      "initialization_options": {
        "usePlaceholders": false,
        "completeFunctionCalls": true
      }
    }
  },
  ```

**Reply by voidtask** [reply]
That's actually exactly what I was trying to achieve, thanks!

**Comment by Blankeos** [comment]
Anyone know how to disable this in Rust as well?

**Reply by Blankeos** [reply]
Claude said:
```jsonc
{
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "completion": {
          "callable": {
            "snippets": "none"
          }
        }
      }
    }
  }
}
```

It works

**Comment by Gutiiii** [comment]
Anyone knows how to do that in Java aswell?
```json
"jdtls": {
			"settings": {
				"initialization_options": {
					"settings": {
						"java": {
							"completion": {
								"guessMethodArguments": "off",
							},
						},
					},
				},
			},
		},
```
		This somehow doesn't seem to work
